split_interleave_fastqs.c

```diff
@@ -8,40 +8,67 @@
 #include <string.h>
 #include <unistd.h>
 
+#define MAX_SUFFIX_LEN 24  // _chunkNNNNNN.fastq
 #define MAX_FILENAME_LEN 4096
 #define DEFAULT_READ_PAIRS_PER_CHUNK 1000000
+#define DEFAULT_TEMP_PREFIX "sample"
```

Comment:
This seems like a footgun, since running two copies of this program at once that don't set a temp prefix will give garbled output.  Why not make it mandatory and warn users that it must be unique across all currently executing processes?